File: python/1071.py
class Solution:
    def gdc(a, b):
        return a if b==0 else gdc(b, a&b)
    def gcdOfStrings(self, str1: str, str2: str) -> str:
        if(str1+str2 != str2+str1):
            return ""
        return str1[0:gcd(len(str1), len(str2))]
        

        

File: cpp/1071.cpp
class Solution {
public:
    int gcd(int a, int b) {
        return b == 0 ? a : gcd(b, a % b); 
    }
    
    string gcdOfStrings(const string& str1, const string& str2) {
        if(str1+str2 != str2+str1) return "";
        return str1.substr(0, gcd(str1.size(), str2.size()));
    }
};